test/e2e/es_cluster_logging.go

```diff
@@ -240,6 +240,18 @@ func ClusterLevelLoggingWithElasticsearch(f *Framework) {
 	missing := 0
 	expected := nodeCount * countTo
 	for start := time.Now(); time.Since(start) < graceTime; time.Sleep(10 * time.Second) {
+
+		// Debugging code to report the status of the elasticsearch logging endpoints.
+		esPods, err := f.Client.Pods(api.NamespaceDefault).List(labels.Set{"k8s-app": "elasticsearch-logging"}.AsSelector(), fields.Everything())
+		if err != nil {
+			Logf("Failed to find any elasticsearch logging nodes: %v", err)
```

Comment:
OK, gotcha.  In that case my suggestion as follows:

1. If it's a real, non-recoverable failure, then Failf(), not Logf() (and hence fail the test without any further retries).
2. If it's a recoverable error, then log it as such (wording similar to my suggestion above).  By all means include the error details in the log message for debugging purposes.